src/js/lib/fileType.js

```diff
@@ -0,0 +1,83 @@
+/* @flow */
+
+import fs from "fs"
+import readline from "readline"
+
+const PCAP_1_HEX = "d4c3b2a1"
+const PCAP_2_HEX = "a1b2c3d4"
+const PCAPNG_HEX = "0a0d0d0a"
+const PCAP_HEXES = [PCAP_1_HEX, PCAP_2_HEX, PCAPNG_HEX]
+
+type IngestableType = "pcap" | "zeek" | "unknown"
+
+export default async function fileType(path: string): Promise<IngestableType> {
+  if (await isPcap(path)) {
+    return "pcap"
+  } else if ((await isZeekAscii(path)) || (await isZeekJson(path))) {
+    return "zeek"
+  } else {
+    return "unknown"
+  }
+}
+
+async function isZeekJson(file) {
+  for await (let line of firstLines(file, 4)) {
+    if (!isJson(line)) return false
+  }
+  return true
+}
+
+async function isZeekAscii(file) {
+  for await (let line of firstLines(file, 4)) {
+    if (!isZeekHeader(line)) return false
+  }
+  return true
+}
+
+async function isPcap(file) {
+  let bytes = await firstBytes(file, 4)
+  for (let hex of PCAP_HEXES) {
+    if (bytes instanceof Buffer && bytes.equals(Buffer.from(hex, "hex")))
+      return true
+  }
+  return false
+}
+
+function isZeekHeader(line) {
+  return /^#\w+\s+\S+/.test(line)
```

Comment:
This will match a trivially-contrived sourcable sh file, like
```sh
#source me to do stuff
export A=b
```
Instead of `^#\w+\s+\S+`, check for `^#separator \S+` ?

Reply:
I see what you mean. How does the proposed regex prevent that problem?

I'm also checking the first 4 lines of the file to see if they match that regex. Is it true that all zeek filees start with the "separator" header?

Lastly, is it a security concern if they spoof this with a non-zeek file? All we do with it pass along the file path to zqd. Whatever they do in zqd will probably throw an error. I don't think anything is able to be "sourced".

Let me ask a zqd dev if this needs to be more robust. @henridf will this zeek detector regex be good enough for ingest?

Reply:
> Is it true that all zeek filees start with the "separator" header?

I think they must since all lines after have to use the separator as defined.

> Lastly, is it a security concern if they spoof this with a non-zeek file? 

Not necessarily. My $0.02 was, if we are doing file detection we can at least block obviously invalid text files before they go all the way through to zqd.

Reply:
Aww I see what you mean. If someone accidentally gives us a file with standard `#` comments at the top of the file (which many files do). Got it. Yeah I'll update that.

Reply:
Thanks. Still happy to hear anyone else's comments on this issue.